util: add --debug-cycle to help debug print

jensen-yan opened the pull request:
it will enable print cycle in DPRINTF,
print tick is difficult to debug, print cycle helps, DPRINTF print like:
  31635: system.cpu.branchPred: [c: 95] DecoupledBPUWithFTB::tick()
c: 95 means cycle, 95 = 31635 / 333, 333 is ticks for 1 cycle in 3GHz, use cpu->clockPeriod() is greater, but not all class have cpu member.

Change-Id: I0ff1f3e3290842e6ba110393705c003616103924

shinezyy reviewed (changes requested):
The reason that original dprintf printing ticks instead of cycles is that not all clockedobjects are in the same clock domain.

But it is much more convenient for CPU guys to use a CPU core clock (this is the reason of this patch).

To avoid magic 333, I suggest to add a new member function  "cpuCycle()" to "clockedObject".

jensen-yan committed:
misc: Add debug cycle option for DPRINTF logging

- scons build/RISCV/gem5.debug --gold-linker -j100 --debug-cycle
- dprintf likes: "333: system.cpu.fetch: [c: 1] Running stage."
- Introduced a new command line option '--debug-cycle' to enable cycle printing in DPRINTF.
- Updated SConscript
- Enhanced DPRINTF macro to include cycle information when DEBUG_SHOW_CYCLES is enabled.
- Added methods to get and set CPU clock period for accurate cycle calculations.
- Now if cpu clocks change to 2GHz, it also works!

Change-Id: Ia5a3753b088076e397ccafb180692e810b2ac50c
